crates/query-engine/translation/src/translation/mutation/translate.rs

```diff
@@ -0,0 +1,253 @@
+//! Translate an incoming `MutationRequest`.
+
+use indexmap::indexmap;
+use std::collections::BTreeMap;
+
+use indexmap::IndexMap;
+use ndc_sdk::models;
+
+use crate::translation::error::Error;
+use crate::translation::helpers::{Env, State, TableNameAndReference};
+use crate::translation::mutation;
+use query_engine_metadata::metadata;
+use query_engine_sql::sql;
+
+/// Translate the incoming MutationOperation to an ExecutionPlan (SQL) to be run against the database.
+pub fn translate(
+    metadata: &metadata::Metadata,
+    operation: models::MutationOperation,
+    collection_relationships: BTreeMap<String, models::Relationship>,
+) -> Result<sql::execution_plan::Mutation, Error> {
+    let env = Env::new(metadata, collection_relationships);
+
+    match operation {
+        models::MutationOperation::Procedure {
+            name,
+            arguments,
+            fields,
+        } => {
+            // lookup native query first
+            match env.lookup_native_query(&name) {
+                Ok(native_query) => {
+                    translate_native_query(&env, name, fields, arguments, native_query)
+                }
+                Err(_) =>
+                // and failing that, try a generated mutation,
+                {
+                    env.lookup_generated_mutation(&name).and_then(|mutation| {
+                        translate_generated_mutation(&env, name, fields, arguments, mutation)
+                    })
+                }
+            }
+        }
+    }
+}
+
+fn translate_generated_mutation(
```

Comment:
The name of this function seems to be a bit off for what it does.

Reply:
Agree, I don't think "generated" is a helpful word to bring with us into the codebase since nearly the whole thing generates something or other.

Reply:
Yeah - also it only produces delete mutations, right?